job-service/server/src/test/java/org/verapdf/webapp/jobservice/server/controller/JobServiceTests.java

```diff
@@ -0,0 +1,206 @@
+		                                                                    .contentType(MediaType.APPLICATION_JSON)
+		                                                                    .content(requestBody))
+		                                     .andExpect(status().isBadRequest())
+		                                     .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON_VALUE))
+		                                     .andExpect(jsonPath("$.error").value(HttpStatus.BAD_REQUEST.getReasonPhrase()))
+		                                     .andExpect(jsonPath("$.message")
+				                                                .value("Invalid arguments"))
+		                                     .andExpect(jsonPath("$.status").value(HttpStatus.BAD_REQUEST.value()))
+		                                     .andExpect(jsonPath("$.timestamp").isNotEmpty())
+		                                     .andReturn().getResolvedException();
+		assertNotNull(resolvedException);
+		assertEquals(MethodArgumentNotValidException.class, resolvedException.getClass());
+		assertTrue(resolvedException.getMessage().contains("Validation failed for argument"));
+	}
+
+	@Test
+	public void createJobWithRedundantDTOFieldTest() throws Exception{
+		String requestBody = "{\"profile\":\"TAGGED_PDF\"," +
+		                     "\"redundantField\":\"redundant\"}";
+		//Creating job
+		MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/job")
+		                                                         .contentType(MediaType.APPLICATION_JSON)
+		                                                         .content(requestBody))
+		                          .andExpect(status().isOk())
+		                          .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+		                          .andExpect(jsonPath("$.id").isNotEmpty())
+		                          .andExpect(jsonPath("$.profile").value("TAGGED_PDF"))
+		                          .andExpect(jsonPath("$.status").value("CREATED"))
+		                          .andExpect(jsonPath("$.sourceFiles").isEmpty()).andReturn();
+	}
+
+	@Test
+	public void getJobWithoutJobInDatabaseTest() throws Exception{
+		String id = "893ce251-4754-4d92-a6bc-69f886ab1ac6";
+		//Retrieving storedFile data
+		Exception resolvedException = mockMvc.perform(MockMvcRequestBuilders.get("/job/" + id)
+		                                                                    .accept(MediaType.APPLICATION_JSON_VALUE))
+		                                     .andExpect(status().isNotFound())
+		                                     .andExpect(jsonPath("$").doesNotExist())
+		                                     .andReturn().getResolvedException();
+		assertNotNull(resolvedException);
+		assertEquals(NotFoundException.class, resolvedException.getClass());
+		assertEquals("Job with specified id not found in DB: " + id, resolvedException.getMessage());
+	}
+}
```

Comment:
missing tests:
1. null value inside source files list
2. (as we want to add ignore null for json) not null error type and error message (with error status)
3. (as we want to add ignore null for json) not null result id (with FINISHED status)

Reply:
4. POST job with id test (the posted id should be ignored)
4. invalid status/other values (not corresponding to enum) in tasks

Reply:
6. add a test with both profile and tasks invalid data (to ensure that we get both fields errors in the response)

Reply:
7. modify a valid test with files (now it has only 1 task) such that it would have 2 tasks